### DISCUSSION TITLE
dwzwave25 : Modify Door/Window sensor to connect to Smoke Detector

### DISCUSSION THREAD

**Discussion by ckbisek** [zwave-js/zwave-js#3551] [Q&A]
I am recently moving from SmartThings where I had a modified "Device Handler" for an Ecolink Door/Window sensor physically connected to a "dumb" smoke/co detector via the interconnect.  

I'm at a loss of how to mimic this functionality

https://github.com/vseven/SmartThings_VSeven/blob/master/devicetypes/vseven/modified-zwave-door-window-sensor-for-smoke.src/modified-zwave-door-window-sensor-for-smoke.groovy

**Comment by AlCalzone** [comment]
Can you point me to something specific in that file or explain a bit more in depth what you're trying to do?

**Comment by ckbisek** [comment]
@AlCalzone 

Thank you for your reply
I am attempting to receive alerts from the smoke detector by means of the zwave sensor which has leads connected to the smoke detector via the "interconnect".  

When smoke is detected, the signal is sent via the interconnect and the zwave sensor changes state.  Essentially, instead of OPEN or CLOSE the sensor would be ALERT or NORMAL

This is the one last function that is keeping me from throwing out the SmartThings hub; yup, I realize I could spend 3-4x the cost for a battery powered smoke/co detector which isn't hardwired ... ;)

**Reply by AlCalzone** [reply]
So you want to re-interpret the switch states as alarm states?
That's not something the driver supports. If you're using Home Assistant, you might be able to realize it with some kind of template sensor. But I can't help with that since I'm not using HA myself.

**Reply by ckbisek** [reply]
@AlCalzone 
Thank you, I'll look at it again.  I interpreted the issue as a communication hurdle that needed to be translated as I could not find any way to change how the switch interacted or reported any change in state when the alarm would go off.  

I'll perform some additional validations; however will likely lose interest ;)

**Reply by AlCalzone** [reply]
Just to be sure, the device **does** report something when you trigger the alarm?

**Reply by ckbisek** [reply]
@AlCalzone 
Nope, upon removing it from SmartThings with "custom handler", I had not been able to see a response when the alarm was triggered.

**Reply by AlCalzone** [reply]
Can you share a [driver log](https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/generating-logs?id=driver-logs) where you trigger the alarm? The linked manual is for zwavejs2mqtt, HomeAssistant gives you similar options AFAIK.
Make sure it is on loglevel `debug` or `silly` and attach it here as a file (by drag&drop).
